1.23 CLI Release Docs (#3388)
astronomer/docs: astro/deployment-resources.md

The [Kubernetes executor](kubernetes-executor.md) and [KubernetesPodOperator](kubernetespodoperator.md) both use Kubernetes Pods to execute tasks. While you still need to configure Pods in your DAG code to define individual task environments, you can set some safeguards on Astro so that tasks in your Deployment don't request more CPU or memory than expected.

Set safeguards by configuring default Pod limits and requests from the Cloud UI. If a task requests more CPU or memory than is currently allowed in your configuration, the task fails.

> ::: info

Reviewer: ```suggestion
:::info
```

> You can also configure default pod limits and resources with the [`astro deployment create`](cli/astro-deployment-create.md) and [`astro deployment update`](cli/astro-deployment-update.md) Astro CLI commands, or by adding the configurations to a [Deployment file](deployment-file-reference.md), if you manage your Deployment as code.

Reviewer: ```suggestion
To manage Kubernetes resources programmatically, you can set default Pod limits and resources with the [`astro deployment create`](cli/astro-deployment-create.md) and [`astro deployment update`](cli/astro-deployment-update.md) Astro CLI commands, or by adding the configurations to a [Deployment file](deployment-file-reference.md).
```

The [Kubernetes executor](kubernetes-executor.md) and [KubernetesPodOperator](kubernetespodoperator.md) both use Kubernetes Pods to execute tasks. While you still need to configure Pods in your DAG code to define individual task environments, you can set some safeguards on Astro so that tasks in your Deployment don't request more CPU or memory than expected.

Set safeguards by configuring default Pod limits and requests from the Cloud UI, through the Astro CLI, or with a Deployment configuration. If a task requests more CPU or memory than is currently allowed in your configuration, the task fails.

<Tabs
    groupId= "create-a-cluster"
    values={[
        {label: 'Cloud UI', value: 'cloud-ui'},
        {label: 'Astro CLI', value: 'astro-cli'},

>         {label: 'Deployment file', value: 'deployment-file'},

Reviewer: This is a really complex structure that, assuming we want it here, would apply to all other resource sections. I assumed that we never documented Deployment file setups or Astro CLI setups here because those were considered automation workflows, and thus discussed in detail within core configuration docs. 

My suggestion here is:
- Add notes to the beginning of the doc to explain that these settings can also be updated using a Deployment file or the Astro CLI.  Then link to https://docs.astronomer.io/astro/manage-deployments-as-code
- Avoid topic-level tabs to simplify this doc
